lib/avromatic/model/types/decimal_type.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'bigdecimal'
+require 'bigdecimal/util'
+require 'avromatic/model/types/abstract_type'
+
+module Avromatic
+  module Model
+    module Types
+      class DecimalType < AbstractType
+        VALUE_CLASSES = [::BigDecimal].freeze
+        INPUT_CLASSES = [::Numeric, ::String].freeze
+        DEFAULT_PRECISION = 3
+        private_constant :DEFAULT_PRECISION
+
+        def value_classes
+          VALUE_CLASSES
+        end
+
+        def input_classes
+          INPUT_CLASSES
+        end
+
+        def name
+          'decimal'
+        end
+
+        def coerce(input)
+          case input
+          when ::NilClass, ::BigDecimal
+            input
+          when ::Rational
+            input.to_d(DEFAULT_PRECISION)
```

Comment:
Should we be truncating `Float`s too?

Reply:
Wouldn't it be better to delegate that to Avro? https://github.com/apache/avro/blob/4e1fefca493029ace961b7ef8889a3722458565a/lang/ruby/lib/avro/logical_types.rb#L180-L200

Reply:
After more consideration it seems that supporting coercion for `Rational` doesn't make a lot of sense here, because eventually it stumbles into the rounding issue anyway. And if I understand correctly the Avro ruby library's approach, they avoid rounding and instead delegate it to the user by raising `Rounding necessary` error when not rounded value provided.

The only benefit of having support for `Rational` coercion is to not raise error when `input.to_d` is called without `precision`, which is [required by specs](https://ruby-doc.org/3.2.1/exts/bigdecimal/Rational.html#method-i-to_d)

Reply:
I agree with your assessment. We're safer following the same approach as Avro Ruby and delegating any rounding decisions to the client. Thus no need to support `Rational`.